src/nupic/utils/Random.cpp

```diff
@@ -66,7 +68,11 @@ std::istream &operator>>(std::istream &inStream, Random &r) {
   NTA_CHECK(version == "random-v2") << "Random() deserializer -- found unexpected version string '"
               << version << "'";
   inStream >> r.seed_;
-  inStream >> r.gen;
+  r.gen.seed(r.seed_); //reseed
+  inStream >> r.steps_;
+  r.gen.discard(r.steps_); //advance n steps
+  //FIXME we could de/serialize directly RNG gen, it should be multi-platform according to standard, 
+  //but on OSX CI it wasn't (25/11/2018). So "hacking" the above instead. 
```

Comment:
serialization of RNG gen should have worked, but OSX had issues. So implemented this manual method: 
it may be slightly slower to deserialize, on the other hand the data stream is much simpler.